tests/ha/single_pod_restart/test_data_server_pod_restart.py

```diff
@@ -518,3 +524,407 @@ def test_io_data_server_pod_restart(self):
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 3: Performed READs/Verify on data written in healthy cluster in step-1")
+
+        self.test_prefix_deg = self.test_prefix
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("Step 4: Create new bucket and perform WRITEs/READs/Verify with variable "
+                        "object sizes in degraded mode")
+            self.test_prefix_deg = f'test-45536-deg-{int(perf_counter_ns())}'
+        LOGGER.info("Step 4: Create new objects and perform WRITEs/READs/Verify with variable "
+                    "object sizes in degraded mode")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix_deg,
+                                                    skipcleanup=True, setup_s3bench=False,
+                                                    nsamples=5, nclients=5)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 4: Performed WRITEs/READs/Verify with variable sizes objects in "
+                    "degraded mode")
+
+        LOGGER.info("Step 5: Shutdown one server pod with replica method and verify cluster & "
+                    "remaining pods status")
+        pod_prefix = const.SERVER_POD_NAME_PREFIX
+        num_replica = self.pod_dict[pod_prefix][-1] - 1
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+            pod_prefix=[pod_prefix], delete_pod=[self.pod_dict.get(pod_prefix)[0]],
+            num_replica=num_replica)
+        assert_utils.assert_true(resp[1], "Failed to shutdown/delete pod")
+        pod_name = list(resp[1].keys())[0]
+        self.pod_dict[pod_prefix].append(resp[1][pod_name]['deployment_name'])
+        self.pod_dict[pod_prefix].append(resp[1][pod_name]['method'])
+        assert_utils.assert_true(resp[0], "Cluster/Services status is not as expected")
+        LOGGER.info("successfully shutdown pod %s", self.pod_dict.get(pod_prefix)[0])
+        self.restore_pod = True
+        LOGGER.info("Step 5: Successfully shutdown one server pod with replica method and verify "
+                    "cluster & remaining pods status")
+
+        LOGGER.info("STEP 6.1: Perform READs/Verify on data written in healthy cluster in step-1")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix, skipwrite=True,
+                                                    skipcleanup=True, setup_s3bench=False,
+                                                    nsamples=5, nclients=5)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 6.1: Performed READs/Verify on data written in healthy cluster in step-1")
+
+        if CMN_CFG["dtm0_disabled"]:
```

Comment:
we don't need this flag here right? As we have already assigned either healthy mode bucket or deg mode bucket to the prefix we can run anyways.